src/util/codec/mysql/json/functions.rs

```diff
@@ -0,0 +1,163 @@
+// Copyright 2017 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// FIXME: remove following later
+#![allow(dead_code)]
+
+use std::{u32, char};
+use super::super::Result;
+use super::json::Json;
+
+const ESCAPED_UNICODE_BYTES_SIZE: usize = 4;
+
+const CHAR_BACKSPACE: char = '\x08';
+const CHAR_HORIZONTAL_TAB: char = '\x09';
+const CHAR_LINEFEED: char = '\x0A';
+const CHAR_FORMFEED: char = '\x0C';
+const CHAR_CARRIAGE_RETURN: char = '\x0D';
+
+impl Json {
+    pub fn unquote(&self) -> Result<String> {
+        match *self {
+            Json::String(ref s) => unquote_string(s),
+            _ => Ok(self.to_string()),
+        }
+    }
+}
+
+// unquote_string recognizes the escape sequences shown in:
+// https://dev.mysql.com/doc/refman/5.7/en/json-modification-functions.html#
+// json-unquote-character-escape-sequences
+pub fn unquote_string(s: &str) -> Result<String> {
+    let mut ret = String::with_capacity(s.len());
+    let mut chars = s.chars();
+    while let Some(ch) = chars.next() {
+        if ch == '\\' {
+            let c = match chars.next() {
+                Some(c) => c,
+                None => return Err(box_err!("Incomplete escaped sequence")),
+            };
+            match c {
+                '"' => ret.push('"'),
+                'b' => ret.push(CHAR_BACKSPACE),
+                'f' => ret.push(CHAR_FORMFEED),
+                'n' => ret.push(CHAR_LINEFEED),
+                'r' => ret.push(CHAR_CARRIAGE_RETURN),
+                't' => ret.push(CHAR_HORIZONTAL_TAB),
+                '\\' => ret.push('\\'),
+                'u' => {
+                    let mut unicode = String::with_capacity(ESCAPED_UNICODE_BYTES_SIZE);
+                    for _ in 0..ESCAPED_UNICODE_BYTES_SIZE {
+                        match chars.next() {
+                            Some(c) => unicode.push(c),
+                            None => return Err(box_err!("Invalid unicode: {}", unicode)),
+                        }
+                    }
+                    let utf8 = try!(decode_escaped_unicode(&unicode));
```

Comment:
I think here we can't `try!`, because if it's `\uBADF`, we should raise the error instead of panic.